README.md

```diff
@@ -50,37 +50,72 @@ const styles = css`
 For those that want something a bit more like styled-components, there is an experimental component API!
 
 ```js
-import { styled, css } from 'css-literal-loader/styled'; // import needed!
+import styled from 'css-literal-loader/styled'; // import needed!
 
 const Button = styled('button')`
   color: black;
   border: 1px solid black;
   background-color: white;
 
-  ${props => props.primary && css`
+  &.primary {
     color: blue;
     border: 1px solid blue;
-  `}
+  }
+
+  &.color.green {
+    color: green;
+  }
 `;
+```
+
+You can render this with:
 
-render(<Button>A styled button</Button>, mountNode);
+```js
+render(<Button primary color="green">A styled button</Button>, mountNode);
 ```
 
-The above transpiles to
+The above transpiles to something like:
 
 ```js
-const Button = styled('div', 'Button', require('./FileName-Button.css'), styles => [
-  styles.button,
-  props => props.primary && styles.buttonVariant1
-]
+const styles = css`
+  .button {
+    color: black;
+    border: 1px solid black;
+    background-color: white;
+
+    &.primary {
+      color: blue;
+      border: 1px solid blue;
+    }
+
+    &.color.green {
```

Comment:
we could do the decss thing of `.color-green` as well... that would definitely require pulling in `lodash/camelCase`, though. also it wouldn't let us delete the prop if the value doesn't match. it would avoid the weird precedence things here, though, and it'd prevent `green="color"` from working

Reply:
+1, `color=green`?

Reply:
it'd have to be `.color-green`, and because of camelcasing things it'd get munged into `.colorGreen`, and we'd support both `color="green"` and `colorGreen={true}` as props, but that's not too important

right now the logic is going through all the props and seeing which ones match classes

Reply:
I like that better, but it's probably not possible without adding a css parser. My only concern there is we have no idea waht flavor of css the user is writing so it's tough to try and parse it (e.g. LESS, SCSS, custom postcss) I don't know that we'd get a reliable AST without user imput, but you'd be the expert there :P

Reply:
yeah so i think i'll just do the dashed class names and pull in `lodash/camelCase`? so you'll do `.theme-green`, but we'll look for `.themeGreen` and `.theme-green` when you do `theme="green"`

Reply:
I like that better, the cost of the lodash function will be small enough